### PR TITLE
Make beam centre results comparible

### LINKED ISSUE #27966: Beam Centre Finder results cannot be compared
Original Reporter: ISIS/Steve

The output in SANS beam centre finder currently accumulates the total difference in each quadrant 
as SX / SY. 

However as we don't average by the number of points considered a better fit over a larger radius will result in a "worse" value, despite converging to a better result.

We should average the results by the number of points, and have been asked to print the number of considered points vs total points

### PR DESCRIPTION
**Pull request by DavidFair**
**Description of work.**
    Change the SANS Beam Centre Finder to prints the average residual change
    between two quarters. This allows for direct comparisons, previously if
    a user increased the radius limit it would appear to perform worse,
    despite actually doing better. This was because the additional number of
    points was not accounted for.

I'd recommend you review the second commit, the first commit (the majority of the diff) split a method call out of PyExec

**Report to:** ISIS/Steve

**To test:**
- Download the ISIS Training Data
- Open the SANS GUI 
- Use `maskfile.txt` in the user file box
- Load `batch_mode_reduction` in the batch file box
- Hit load
- Switch to Beam Centre on the left
- Hit run, check the value are well formatted and readable

Fixes #27966 

---

#### Reviewer ####

Please comment on the following ([full description](http://developer.mantidproject.org/ReviewingAPullRequest.html)):

##### Code Review #####

- Is the code of an acceptable quality?
- Does the code conform to the [coding standards](http://developer.mantidproject.org/Standards/)?
- Are the unit tests small and test the class in isolation?
- If there are changes in the release notes then do they describe the changes appropriately?

##### Functional Tests #####

- Do changes function as described? Add comments below that describe the tests performed?
- Do the changes handle unexpected situations, e.g. bad input?
- Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.

### CODE REVIEW

**Review by alicerussell1 — CHANGES_REQUESTED** [review]
This works well and the code looks good, just one comment below but this may be from code in one of the earlier PR's.

File: qt/python/mantidqt/plotting/test/test_tiledplots.py [hunk 1]
```diff
@@ -97,10 +97,12 @@ def test_get_plot_fig_with_no_figure_provided_and_overplot_is_true_returns_curre
...
         set_figure_as_current_figure(initial_fig)
 
+        # If not figure is provided we provide a user a new figure plot, since this allows
+        # overplotting in a loop, which requires there to be a figure with an associated axis
         fig, axes = get_plot_fig(overplot=True)
 
-        self.assertEqual(fig, initial_fig)
-        self.assertEqual(0, len(fig.axes))
+        self.assertNotEqual(fig, initial_fig)
+        self.assertNotEqual(0, len(fig.axes))
```
[the remark this answers is not in the document]
Comment (DavidFair) [reply qt/python/mantidqt/plotting/test/test_tiledplots.py]: Fixed in this PR to save me affecting 3 others 
Thanks for spotting that

**Review by alicerussell1 — DISMISSED** [review]
The looks good to me 👍

**Comment by gemmaguest** [comment]
@DavidFair There are conflicts

**Review by alicerussell1 — APPROVED** [review]
Still works 👍

**Pull request merged**